applications/ThermalDEMApplication/custom_constitutive/generation/generation_dissipation.cpp

```diff
@@ -35,34 +35,64 @@ namespace Kratos {
     // Conversion and partition coefficients
     const double conversion = r_process_info[HEAT_GENERATION_RATIO];
     const double partition  = ComputePartitionCoeff(particle);
+    const double coeff      = conversion * partition;
 
     // Initialize contribution from different sources of heat generation
-    double heat_gen_damping = 0.0;
-    double heat_gen_sliding = 0.0;
-    double heat_gen_rolling = 0.0;
+    double heat_gen;
+    double heat_gen_damping_pp = 0.0;
+    double heat_gen_damping_pw = 0.0;
+    double heat_gen_sliding_pp = 0.0;
+    double heat_gen_sliding_pw = 0.0;
+    double heat_gen_rolling_pp = 0.0;
+    double heat_gen_rolling_pw = 0.0;
 
     // Damping thermal power
     if (r_process_info[GENERATION_DAMPING_OPTION]) {
-      heat_gen_damping = partition * conversion * contact_params.viscodamping_energy / time;
-      if      (particle->mNeighborType & PARTICLE_NEIGHBOR) particle->mGenerationHeatFlux_damp_particle += heat_gen_damping;
-      else if (particle->mNeighborType & WALL_NEIGHBOR)     particle->mGenerationHeatFlux_damp_wall     += heat_gen_damping;
+      heat_gen = coeff * contact_params.viscodamping_energy / time;
+
+      if (particle->mNeighborType & PARTICLE_NEIGHBOR) {
+        heat_gen_damping_pp = heat_gen;
+        particle->mGenerationHeatFlux_damp_particle += heat_gen;
+      }
+      else if (particle->mNeighborType & WALL_NEIGHBOR) {
+        heat_gen_damping_pw = heat_gen;
+        particle->mGenerationHeatFlux_damp_wall += heat_gen;
+      }  
     }
 
     // Sliding friction thermal power
     if (r_process_info[GENERATION_SLIDING_OPTION]) {
-      heat_gen_sliding = partition * conversion * contact_params.frictional_energy / time;
-      if      (particle->mNeighborType & PARTICLE_NEIGHBOR) particle->mGenerationHeatFlux_slid_particle += heat_gen_sliding;
-      else if (particle->mNeighborType & WALL_NEIGHBOR)     particle->mGenerationHeatFlux_slid_wall     += heat_gen_sliding;
+      heat_gen = coeff * contact_params.frictional_energy / time;
+
+      if (particle->mNeighborType & PARTICLE_NEIGHBOR) {
```

Comment:
these operations would be performed inside this new class object initialized at the very begining of the simulation

Reply:
I didn't get it.